nixarr/monitoring/default.nix

```diff
@@ -0,0 +1,328 @@
+          description = "Port for Prowlarr metrics";
+        };
+        listenAddr = mkOption {
+          type = types.str;
+          default = "127.0.0.1";
+          description = "Address for Prowlarr exporter to listen on";
+        };
+      };
+    };
+  };
+
+  config = mkIf (cfg.enable && cfg.exporters.enable) {
+    # Configure Prometheus exporters
+    services.prometheus = {
+      exporters = {
+        # Enable exportarr for each supported service if it's enabled
+        exportarr-sonarr =
+          mkIf (
+            cfg.sonarr.enable
+            && (cfg.sonarr.exporter.enable == null || cfg.sonarr.exporter.enable)
+          ) {
+            enable = true;
+            url = "http://127.0.0.1:8989";
+            apiKeyFile = "${cfg.stateDir}/api-keys/sonarr.key";
+            port = cfg.sonarr.exporter.port;
+            listenAddress = cfg.sonarr.exporter.listenAddr;
+          };
+
+        exportarr-radarr =
+          mkIf (
+            cfg.radarr.enable
+            && (cfg.radarr.exporter.enable == null || cfg.radarr.exporter.enable)
+          ) {
+            enable = true;
+            url = "http://127.0.0.1:7878";
+            apiKeyFile = "${cfg.stateDir}/api-keys/radarr.key";
+            port = cfg.radarr.exporter.port;
+            listenAddress = cfg.radarr.exporter.listenAddr;
+          };
+
+        exportarr-lidarr =
+          mkIf (
+            cfg.lidarr.enable
+            && (cfg.lidarr.exporter.enable == null || cfg.lidarr.exporter.enable)
+          ) {
+            enable = true;
+            url = "http://127.0.0.1:8686";
+            apiKeyFile = "${cfg.stateDir}/api-keys/lidarr.key";
+            port = cfg.lidarr.exporter.port;
+            listenAddress = cfg.lidarr.exporter.listenAddr;
+          };
+
+        exportarr-readarr =
+          mkIf (
+            cfg.readarr.enable
+            && (cfg.readarr.exporter.enable == null || cfg.readarr.exporter.enable)
+          ) {
+            enable = true;
+            url = "http://127.0.0.1:8787";
+            apiKeyFile = "${cfg.stateDir}/api-keys/readarr.key";
+            port = cfg.readarr.exporter.port;
+            listenAddress = cfg.readarr.exporter.listenAddr;
+          };
+
+        exportarr-prowlarr =
+          mkIf (
+            cfg.prowlarr.enable
+            && (cfg.prowlarr.exporter.enable == null || cfg.prowlarr.exporter.enable)
+          ) {
+            enable = true;
+            url = "http://127.0.0.1:9696";
+            apiKeyFile = "${cfg.stateDir}/api-keys/prowlarr.key";
+            port = cfg.prowlarr.exporter.port;
+            listenAddress = cfg.prowlarr.exporter.listenAddr;
+          };
+
+        # Enable node and systemd exporters by default
+        node = {
+          enable = true;
+          enabledCollectors = ["systemd" "tcpstat" "network_route"];
+        };
+        systemd.enable = true;
+
+        # Configure wireguard exporter
+        wireguard = mkIf cfg.vpn.enable {
```

Comment:
What does this do? Doesn't this just run the service through the vpn?

Reply:
This collects some wireguard stats such as last handshake time. I have yet to test it.

Reply:
Hmm, do you want to? I don't care too much. In any case this is approved when you think it's done and the CI is pleased. Oh, forgot about the changelog, could you add this? Otherwise everything looks good. I'll dogfood this for a bit on my machine (dev branch) and ping you on discord if there's any major issues. Sound good. After a while I'll merge to main.

Reply:
Yes, want to, just been busy :) will do it later, and also add a flag to enable metrics for autobrr. I'll ping you later!

Reply:
Tested it and it works, however for some reason I cannot access the port from the host. Needs more investigation.

Reply:
> Tested it and it works, however for some reason I cannot access the port from the host. Needs more investigation.

This is probably some VPN nonsense. It only doesn't work if it's `localhost`?

Reply:
To my knowledge it's gotta be in the same network namespace, hence I'm using the VPN option (I could set it manually too, but I figured it's more fragile). I'll try once more.